File: server/internal/model/subtask.go
package model

import (
	"time"
)

// Subtask 子任务模型
type Subtask struct {
	ID          uint      `json:"id" gorm:"primaryKey"`
	TaskID      uint      `json:"task_id" gorm:"not null;index"`
	Title       string    `json:"title" gorm:"size:500;not null"`
	IsCompleted bool      `json:"is_completed" gorm:"default:false"`
	SortOrder   int       `json:"sort_order" gorm:"default:0"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

File: client/lib/models/subtask.dart
class Subtask {
  final int id;
  final int taskId;
  final String title;
  final bool isCompleted;
  final int sortOrder;
  final DateTime createdAt;

  Subtask({
    required this.id,
    required this.taskId,
    required this.title,
    this.isCompleted = false,
    this.sortOrder = 0,
    required this.createdAt,
  });

  factory Subtask.fromJson(Map<String, dynamic> json) {
    return Subtask(
      id: json['id'],
      taskId: json['task_id'],
      title: json['title'],
      isCompleted: json['is_completed'] ?? false,
      sortOrder: json['sort_order'] ?? 0,
      createdAt: DateTime.parse(json['created_at']),
    );
  }
}
